AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m

```diff
@@ -62,6 +64,13 @@ - (NSURLRequest *)createRequest:(NSObject *)data eTag:(NSString *)__unused eTag
   request.allHTTPHeaderFields = self.httpHeaders;
   [request setValue:self.appSecret forHTTPHeaderField:kMSHeaderAppSecretKey];
 
+  // Copy self.authToken into a local variable to avoid a race condition.
+  NSString *authTokenCopy = self.authToken;
```

Comment:
is the property marked as atomic then? if not need to access it via synchronized?

Reply:
Ah yes it's atomic in the header, all good